packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/extension/model/extensionmodel/PageableContinuationToken.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.typespec.http.client.generator.core.extension.model.extensionmodel;
+
+import com.azure.json.JsonReader;
+import com.azure.json.JsonSerializable;
+import com.azure.json.JsonWriter;
+import com.microsoft.typespec.http.client.generator.core.extension.base.util.JsonUtils;
+import com.microsoft.typespec.http.client.generator.core.extension.model.codemodel.Header;
+import com.microsoft.typespec.http.client.generator.core.extension.model.codemodel.Parameter;
+import com.microsoft.typespec.http.client.generator.core.extension.model.codemodel.Property;
+import java.io.IOException;
+import java.util.List;
+
+public class PageableContinuationToken implements JsonSerializable<PageableContinuationToken> {
```

Comment:
for my own understanding - is this type (or any types generally under the extensionmodel namespace) intended to be utilized by the final generated SDKs that we ship?

Reply:
It actually be classes under `codemodel` and `extensionmodel`.

Most of them is still from AutoRest M4 code-model. We are adding some for TypeSpec (like this class, and involve some structural changes like supporting multiple client), but the basic is still code-model.

And yes, the idea is these classes are what upstream think codegen need to consider when generate Java code. Though codegen can ignore some, e.g. currently we don't have logic to support this continuationToken in Azure SDK output.